### PR TITLE
SB-46: Add a create reservation feature

### PR DESCRIPTION
**Pull request by lubualo**
- A create reservation method is created in the reservation service that communicates with the firebase database.
- A Create Reservation page was created just to test this.

No screenshot seemed relevant.

### CODE REVIEW

**Comment by lubualo** [comment]
[Get/Set Court Reservations in Firebase](https://trello.com/c/T0QhLWBt/46-get-set-court-reservations-in-firebase)

File: app/createReservation/page.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+			}}
+			handleSubmit={(data: Reservation): Promise<boolean> => {
+				const db = getFirestore(firebaseApp);
+				const docRef = doc(
+					db,
+					process.env.NEXT_PUBLIC_COURT_COLLECTION ??
+						'NEXT_PUBLIC_COURT_COLLECTION',
+					process.env.NEXT_PUBLIC_BURATO_COURT_ID ??
+						'NEXT_PUBLIC_BURATO_COURT_ID'
+				);
```
Comment (EzeLamar) [inline app/createReservation/page.tsx:26]: Although this page will be used for testing the Reservation CRUD, the firebase logic must be managed by the firebase service.

Create a new function as follows in `firebase/courts`: 
```javascript
export function getCourtRef() {
	const db = getFirestore(firebaseApp);

	return doc(
		db,
		process.env.NEXT_PUBLIC_COURT_COLLECTION ?? 'NEXT_PUBLIC_COURT_COLLECTION',
		process.env.NEXT_PUBLIC_BURATO_COURT_ID ?? 'NEXT_PUBLIC_BURATO_COURT_ID'
	);
}
```
and use it here.

File: app/createReservation/page.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+				return createReservation({
+					...data,
+					startTime: new Date(data.startTime),
+					endTime: new Date(data.endTime),
+					court: docRef,
+				});
+			}}
```
Comment (EzeLamar) [inline app/createReservation/page.tsx:33]: Also is important to show a toast with a message ( success or error message) to know that the reservation was created or not. 

```javascript
try {
					const result = createReservation(...);
					toast.success('Evento creado!', {
						theme: 'colored',
					});
					return result;
				} catch (error: unknown) {
					if (hasErrorMessage(error)) {
						toast.error(error.message, { theme: 'colored' });
					}

					throw error;
				}
```

File: app/createReservation/page.tsx (hunk 1 above)
Comment (lubualo) [reply app/createReservation/page.tsx]: Done

File: app/createReservation/page.tsx (hunk 2 above)
Comment (lubualo) [reply app/createReservation/page.tsx:58]: Done

**Review by EzeLamar — APPROVED** [review]
Amazing job! 🚀

**Pull request merged**